src/flamenco/runtime/program/fd_vote_program.c

```diff
@@ -39,868 +30,154 @@
-
-    /* Emplace new vote state into target */
-    self->discriminant = fd_vote_state_versioned_enum_current;
-    self->inner.current = current;
-
-    break;
-  }
-  case fd_vote_state_versioned_enum_current:
-    break;
-  default:
-    FD_LOG_ERR(( "unsupported vote state version: %u", self->discriminant ));
-  }
-}
-
-/**********************************************************************/
-/* impl VoteState                                                     */
-/**********************************************************************/
-
-// https://github.com/anza-xyz/agave/blob/v2.0.1/sdk/program/src/vote/state/mod.rs#L428
-static void
-vote_state_new( fd_vote_init_t *              vote_init,
-                fd_sol_sysvar_clock_t const * clock,
-                uchar *                       authorized_voters_mem,
-                fd_vote_state_t *             vote_state /* out */ ) {
-  vote_state->node_pubkey = vote_init->node_pubkey;
-  vote_state->authorized_voters = *authorized_voters_new( clock->epoch, &vote_init->authorized_voter, authorized_voters_mem );
-  // https://github.com/anza-xyz/agave/blob/v2.0.1/sdk/program/src/vote/state/mod.rs#L431
-  vote_state->authorized_withdrawer = vote_init->authorized_withdrawer;
-  vote_state->commission            = vote_init->commission;
-  // https://github.com/anza-xyz/agave/blob/v2.0.1/sdk/program/src/vote/state/mod.rs#L434
-  vote_state->prior_voters.idx      = 31;
-  vote_state->prior_voters.is_empty = 1;
-}
-
-// https://github.com/anza-xyz/agave/blob/v2.0.1/programs/vote/src/vote_state/mod.rs#L985
-static inline int
-verify_authorized_signer( fd_pubkey_t const * authorized,
-                          fd_pubkey_t const * signers[static FD_TXN_SIG_MAX] ) {
-  // https://github.com/anza-xyz/agave/blob/v2.0.1/programs/vote/src/vote_state/mod.rs#L989
-  return fd_signers_contains( signers, authorized ) ?
-    FD_EXECUTOR_INSTR_SUCCESS :
-    FD_EXECUTOR_INSTR_ERR_MISSING_REQUIRED_SIGNATURE;
-}
-
-// lambda function: https://github.com/anza-xyz/agave/blob/v2.0.1/programs/vote/src/vote_state/mod.rs#L873
-static inline int
-verify( fd_pubkey_t *       epoch_authorized_voter,
-        int                 authorized_withdrawer_signer,
-        fd_pubkey_t const * signers[static FD_TXN_SIG_MAX] ) {
-  if( authorized_withdrawer_signer )
-    return 0;
-  else
-    return verify_authorized_signer( epoch_authorized_voter, signers );
-}
-
-// https://github.com/anza-xyz/agave/blob/v2.0.1/sdk/program/src/vote/state/mod.rs#L845
-static void
-pop_expired_votes( fd_vote_state_t * self, ulong next_vote_slot ) {
-  while( !deq_fd_landed_vote_t_empty( self->votes ) ) {
-    fd_landed_vote_t * vote = deq_fd_landed_vote_t_peek_tail( self->votes );
-    if( !( is_locked_out_at_slot( &vote->lockout, next_vote_slot ) ) ) {
-      deq_fd_landed_vote_t_pop_tail( self->votes );
-    } else {
-      break;
+      return FD_EXECUTOR_INSTR_SUCCESS;
     }
+    default:
+      __builtin_unreachable();
```

Comment:
The unreachable can have nasty UB in production builds (e.g. jumping to a random place in the code). They should only be used for performance-critical parts of the code. Let's put FD_LOG_CRIT here instead.